pkg/sql/opt/indexrec/rec.go

```diff
@@ -271,11 +271,9 @@ func (ir *indexRecommendation) constructIndexRec(ctx context.Context) (Rec, erro
 		sb.WriteByte(';')
 		return Rec{sb.String(), TypeCreateIndex}, nil
 	case TypeReplaceIndex:
-		dropCmd := tree.DropIndex{
-			IndexList: []*tree.TableIndexName{{
-				Table: tableName,
-				Index: tree.UnrestrictedName(existingIndex.Name()),
-			}},
+		setCmd := tree.SetVar{
+			Name:   "autocommit_before_ddl",
+			Values: tree.Exprs{tree.NewStrVal("true")},
```

Comment:
nit: true is already the default in 25.1. so i think we should only include this part for backports to 24.3 and earlier.